### ISSUE TITLE
Insert can.compute into can.Map

### ISSUE THREAD

**Issue by Aldredcz** [bitovi/canjs#1409]
Hello,

One feature I would really appreciate - insert can.compute (the value one, not the function one) into can.Map, so I can use it with normal can.Map interface and having all bindings working on 'both sides', as it works when inserting can.Map into other can.Map, example:

```javascript
var slaveMap = new can.Map({foo: 1});
var masterMap = new can.Map(innerMap: slaveMap);

// then this works
masterMap.innerMap.attr('foo', 5);
slaveMap.attr('foo') === 5; // true


var compute = can.compute(1);
var map = new can.Map(insertedCompute: compute);

// how it works
map.insertedCompute(); // yields 1
map.insertedCompute(5); // sets 5

// how it would be awesome
map.attr('insertedCompute') // yields 1
map.attr('insertedCompute', 5) // sets 5
```

What is the main goal? To get this pattern working:
```javascript
var map1 = new can.Map({a: 1, b: 2});
var map2 = new can.Map({c: 3, d: 4});

var map3 = new can.Map({
    a: map1.compute('a'),
    c: map2.compute('c')
});
```

Since Components are based on data coupling it would be extremely useful to have this possibility.

Unfortunately, there would be problem with backward compatibility, since someone could use it nowadays with current functionality, but it could be done by defining method like:

```javascript
var compute = can.compute(1);
var map = new can.Map({});

map.insertCompute('nameOfAttr', compute);
```

**Comment by justinbmeyer** [comment]
Thanks for submitting this. 

I think this might be possible currently with the define plugin. I don't think we can support it natively. I'm not a fan of calling it insertCompute. 

It seems you are just wanting to 2-way bind a compute to a map. I'm not sure why this would help components as they already make this happen.

**Comment by justinbmeyer** [comment]
With the define plugin:

```
var MyMap = can.Map.extend({
  define: {
    nameOfAttr: {
      get: function(val){
        return val || val()
      }
    }
  }
});

new MyMap({ nameOfAttr: can.compute(5) })
```

**Comment by Aldredcz** [comment]
Not sure, the define 'hack' is the case i want achieve.

The case with Components: 

Imagine, you have a Component used by many types of parent Components (lets call it Widget). This Component accepts 3 parameters and modifies them. Since you can't pass computes directly into Component, you probably pass can.Map like:
```javascript
can.Map({
    param1: value,
    param2: value,
    param3: value
});
```
One Widget has all these parameters in one object, so you can just pass it as it is. 

```javascript
var stuff = new can.Map({
    // many params here including our 3
    param1: value,
    param2: value,
    param3: value
    // and many many more
})
```

But another Widget using the same Component could have these parameters spread over many different objects.

In this situation, this would be really good solution:
```javascript
var params = new can.Map({
    param1: obj1.compute('param1'),
    param2: obj2.compute('param2'),
    param3: obj3.compute('notEvenNamedParam3')
});
```
And any change made would also be propagated to the original obj1/obj2/obj3 can.Map.

About syntax, one more idea, what about extending can.compute prototype with a method, which would produce same compute, but when passing to can.Map, it would have described functionality?
```javascript
var params = new can.Map({
    param1: obj1.compute('param1').methodName() // e.g. insertable?
});
```

**Comment by justinbmeyer** [comment]
Sorry, I'm not sure what you are saying. It's confusing because it's abstract, and you're showing map code, not components and views that assemble the behavior you are looking for. 


I'm not sure what you mean by "you probably pass can.Map". You normally pass each "param" individually as an attribute on the component. Ex <my-widget param1="{parent1Key}" ....>. 

> But another Widget using the same Component could have these parameters spread over many different objects.
> 

Yep, but it could pass them accordingly: <my-widget param1="{parent2.someObj.key}" ....>. 

Hit me up on gitter Monday.

**Comment by justinbmeyer** [comment]
@Aldredcz Any more thoughts on this?  By gitter, I meant: https://gitter.im/bitovi/canjs .  Thanks!

**Comment by daffl** [comment]
This should be fixed now via #1486 (as mentioned in #1470)
